assets/model_monitoring/components/src/feature_importance_metrics/feature_importance_utilities.py

```diff
@@ -28,6 +28,40 @@ def convert_pandas_to_spark(pandas_data):
     return spark.createDataFrame(pandas_data)
 
 
+def is_categorical_column(baseline_data, column_name):
+    """Determine whether the column is categorical.
+
+    :param column_name: the column to determine
+    :type column_name: string
+    :param baseline_data: The baseline data meaning the data used to create the
+    model monitor
+    :type baseline_data: pandas.DataFrame
+    :rtype: boolean
+    """
+    baseline_column = pd.Series(baseline_data[column_name])
+    baseline_column_type = baseline_column.dtype.name
+    if (pd.api.types.is_float_dtype(baseline_column) or
+        pd.api.types.is_datetime64_ns_dtype(baseline_column) or
+        pd.api.types.is_timedelta64_ns_dtype(baseline_column)):
+        return False
+    # treat all datetime types as categorical since LightGBM cannot accept anything but bool, int and float
+    if (pd.api.types.is_object_dtype(baseline_column) or pd.api.types.is_string_dtype(baseline_column)
+            or baseline_column_type == "bool" or pd.api.types.is_datetime64_dtype(baseline_column) or
+            pd.api.types.is_timedelta64_dtype(baseline_column)):
+        return True
+    if pd.api.types.is_integer_dtype(baseline_column):
+        distinct_column_values = len(baseline_column.unique())
+        total_column_values = len(baseline_column)
+        distinct_value_ratio = distinct_column_values / total_column_values
+        if distinct_value_ratio < 0.05:
+            return False
+        else:
+            return True
+    # Log the datatype detected and default to true
+    log_time_and_message(f"Unknown column type: {baseline_column_type}")
```

Comment:
so for all the unknown type of column, we set it as categorical column?

Reply:
yes, there shouldn't be too many of these but lightgbm (the model we use for calculating the feature importances) only really can "understand" simple datatypes ie float, int and all other data types will need to be marked as categorical so that lightgbm will do the conversion of the column to int